simulation/groups.py

```diff
@@ -0,0 +1,342 @@
+__all__ = [
```

Comment:
'SGroup',
    'IGroup',
    'RGroup',
    'DGroup'

are the same in implementation? In that case why we even bother to separate them into different classes? Now I see a lot of code duplication.
Please consider if we can do it better. The main goal here is to have one class per one SIR behavior. Do S class instance can be dead? Or do I class instance can be infected once again? Or R class instance can be infected?
Right now I can see the same functionality per every class, in the end, I want to see different functionality per different class. If this is not clear, we can discuss it on Monday morning.